### DISCUSSION TITLE
Implementing a Dynamic Custom Timeout Relative to Base Unmutated Test Duration

### DISCUSSION THREAD

**Discussion by ASuciuX** [sourcefrog/cargo-mutants#276] [Ideas]
Based on our experience, particularly with large packages, we've found it beneficial to use a custom timeout instead of the default one, due to instances where a timeout is necessary. In our Github workflow implementation, we categorize packages as either `big` or `small`. For larger packages, we could initially set a fixed timeout based on current test times, but this becomes problematic as the package evolves. As new features are added, test times are likely to extend, potentially leading to premature timeout terminations. To counter this, we're considering implementing a dynamic scaling factor, such as `1.2x` or `1.5x` the base test time. This approach ensures that as the test duration increases with additional features, the allowable time before a timeout also increases proportionally. I'm currently working on integrating this feature and will submit a pull request soon.

If you have any specific recommendations or insights on this matter, please feel free to share them.

**Comment by sourcefrog** [comment]
Sounds good, thanks for the PR (#277).

This will be in a near future release.
